pokemonterminal/main.py

```diff
@@ -110,28 +69,41 @@ def main(argv=None):
         print("Dry run, exiting.")
         return
 
-    if options.clear:
-        if PIPE_EXISTS:
-            pipe_out = os.open(PIPE_PATH, os.O_WRONLY)
-            os.write(pipe_out, b"quit\n")
-            os.close(pipe_out)
-        scripter.clear_terminal()
-        return
-
-    if is_slideshow and options.id <= 0 and size > 1:
-        if os.name == 'nt':
-            print("Slideshow not supported on Windows yet.")
-            sys.exit(0)
-        if PIPE_EXISTS:
-            print("Slideshow already running in this instance!")
-            sys.exit(0)
-        if options.slideshow <= 0:
-            print("Time has to be greater then 0. You can use decimal values.")
+    e = create_named_event("Pokemon-Terminal_Wallpaper" if options.wallpaper else "Pokemon-Terminal_Terminal")
+    try:
+        if options.clear:
+            if e.is_duplicate():
+                e.set()
+            scripter.clear_terminal()
             return
-        target_func = scripter.change_wallpaper if options.wallpaper else \
-            scripter.change_terminal
-        slideshow(Filter.filtered_list, options.slideshow, target_func)
-        return
+
+        if is_slideshow and options.id <= 0 and size > 1:
+            if e.is_duplicate():
+                print("One or more slideshows is already running.\n")
+                while True:
+                    print("[S]top the previous slideshow(s) / ", end='')
```

Comment:
i don't know if we really care about this case.
If a slideshow is running, and a user asks for a new slideshow, I think they would expect it to just take over. Having the option to abort starting the new slideshow is ok, but I wouldn't expect to use the new slideshow command just to stop the old one. Maybe that's just me.

Reply:
A user might want to replace the slideshow in the current terminal to change the pokemon selection. Furthermore, you can only have one wallpaper slideshow.

Reply:
Sounds like I misunderstood the options. I thought Ignore and continue was replace the slideshow in the current terminal.
If that's not the case then what is the difference? (Maybe the descriptions need to be updated or users will probably have the same confusion)

Reply:
Ignore and continue just trusts ahead, no matter if the other slideshow is already running in the current terminal. If it is actually running in the current terminal, it might produce weird results.

Reply:
Then does it make sense to have that case, if it might just do weird things?
Should we just have options for what we have defined behaviour for: abort current command, and replace slideshow?

I don't really like the idea of supporting (even providing opportunities to the user) to end up with undefined behaviour

Reply:
Because if the old slideshow is not in the current terminal it just makes it run in the current terminal, independently from the old (when the terminal emulator does supports different backgrounds per terminal). There's no reliable and not highly platform specific way to detect the parent terminal and make it a proper feature.